department-of-veterans-affairs/vets-website
# Use 'ssnOrVaFileNumberUI' component pattern for veteran id

## Summary
On the Pension Benefits application(21P-527EZ),
Update Chapter 1 - Applicant information

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Select 'Start your application without signing in'
3. Verify that either the SSN or VA File Number are required to Continue
4. Verify that the SSN and VA file number are validated

## Related issue(s)
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/70328)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots
_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

### Desktop
| Applicant Information | SSN(Error) | VA File Number(Error) | SSN(Valid) | VA File Number(Valid) |
| ------ | ------ | ------ | ------ | ------ |
|![localhost_3001_pension_application_527EZ_applicant_information](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/a91d76c2-d3bb-48ec-871f-995a177c50e2)|![localhost_3001_pension_application_527EZ_applicant_information (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/693ee530-088c-48b2-90dd-1c9ca46dffb3)|![localhost_3001_pension_application_527EZ_applicant_information (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/902e8b88-d22b-43f2-833b-d6729540f4bc)|![localhost_3001_pension_application_527EZ_applicant_information (3)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/33c3bfd9-b249-4b54-9b3d-a5ab45dafe0c)|![localhost_3001_pension_application_527EZ_applicant_information (4)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/e68dc182-8adc-4d31-a02c-34c72736e9e2)


## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/70328)

### Quality Assurance & Testing

### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
